examples/bri-3/src/bri/workgroup/worksteps/agents/workstepsStorage.agent.ts

```diff
@@ -2,92 +2,79 @@ import { Injectable, NotFoundException } from '@nestjs/common';
 import { PrismaService } from '../../../../../prisma/prisma.service';
 import { NOT_FOUND_ERR_MESSAGE } from '../api/err.messages';
 import { Workstep } from '../models/workstep';
+import { getType } from 'tst-reflect';
+import Mapper from '../../../utils/mapper';
 
 // Storage Agents are the only places that talk the Prisma language of models.
 // They are always mapped to and from domain objects so that the business layer of the application
 // does not have to care about the ORM.
 @Injectable()
 export class WorkstepStorageAgent extends PrismaService {
   async getWorkstepById(id: string): Promise<Workstep> {
-    const workstepModel = await this.workstep.findUnique({ where: { id: id } });
+    const workstepModel = await this.workstep.findUnique({ where: { id } });
 
     if (!workstepModel) {
       throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
     }
 
-    return new Workstep( // TODO: Write generic mapper prismaModel -> domainObject
-      workstepModel.id,
-      workstepModel.name,
-      workstepModel.version,
-      workstepModel.status,
-      workstepModel.workgroupId,
-      workstepModel.securityPolicy,
-      workstepModel.privacyPolicy,
-    );
+    return Mapper.map(workstepModel, getType<Workstep>()) as Workstep;
   }
 
   async getAllWorksteps(): Promise<Workstep[]> {
     const workstepModels = await this.workstep.findMany();
-    return workstepModels.map((w) => {
-      return new Workstep(
-        w.id,
-        w.name,
-        w.version,
-        w.status,
-        w.workgroupId,
-        w.securityPolicy,
-        w.privacyPolicy,
-      );
+    return workstepModels.map((workstepModel) => {
+      return Mapper.map(workstepModel, getType<Workstep>()) as Workstep;
     });
   }
 
   async createNewWorkstep(workstep: Workstep): Promise<Workstep> {
+    const {
+      id,
+      name,
+      version,
+      status,
+      workgroupId,
+      securityPolicy,
+      privacyPolicy,
+    } = workstep;
```

Comment:
workstep is already containing all the params, no need to introduce these new lines and reduce readability.

Reply:
will be replaced with one-liner in next commit